src/streamer.rs

```diff
@@ -442,6 +442,39 @@ fn process_blob(
     }
 }
 
+fn validate_blob_against_window(
+    debug_id: u64,
+    pix: u64,
+    consumed: u64,
+    received: u64,
+) -> Option<u64> {
+    // Prevent receive window from running over
+    if pix >= consumed + WINDOW_SIZE {
+        debug!(
```

Comment:
Does it bother anyone that this information is lost if not logging in debug mode? I see the original code did this, so I'm not going to ask you to clean that up here, but could you add a ticket to add an error type and change this function from `Option<u64>` to `Result<u64>`?

Reply:
Sure, will do.